lib/event_sourcery/aggregate_root.rb

```diff
@@ -124,8 +124,12 @@ def load_history(events)
 
     def apply_event(event_class, options = {})
       event = event_class.new(**options.merge(aggregate_id: id))
-      mutate_state_from(event)
-      @changes << event
+      if event.valid?
```

Comment:
I thought about doing that, but was concerned about whether that would leave the aggregate in a bad state. It would think the event had been applied, when in fact it could not be.